packages/hdwallet-metamask/src/metamask.ts

```diff
@@ -78,8 +77,13 @@ export class MetaMaskHDWallet implements core.HDWallet, core.ETHWallet {
     return Promise.resolve("MetaMask");
   }
 
-  public initialize(): Promise<any> {
-    // TODO: Should we perform setup here or in the constructor?
+  public async initialize(): Promise<any> {
+    try {
+      this.provider = await detectEthereumProvider({ mustBeMetaMask: true, silent: false, timeout: 3000 });
+    } catch (e) {
+      console.error(e);
+    }
```

Comment:
IMO we shouldn't eat errors at this layer; web can swallow it if it wants to, but if there's no provider hdwallet itself should complain loudly.